File: app/controllers/whatsit_controller.rb
class WhatsitController < ApplicationController

  # GET /subjects or /subjects.json

  def index
    @whatsit = Whatsit.new
    @results = ''
  end

  def new
    @whatsit = Whatsit.new
    #extract params 
    @whatsit.params = {search:params[:search],action_type:params[:action_type],confirm:params[:confirm],resp:params[:resp]}

    puts "WI #{@whatsit.inspect}"
    # puts "Ruby PARAMS #{params[:search]}  QUERY #{@results.inspect} resp #{ @whatsit.inspect}"
  end
  def whatsit_params
    params.require(:whatsit).permit(:params)
  end

end

File: app/javascript/controllers/whatsit_controller.js
import { Controller } from "stimulus"
import Rails from '@rails/ujs';

export default class extends Controller {

  static targets = ["console",'query','input_node']

  connect() {
    this.moveCursor()
  }

  search(){
    const tuple_node = event.target
    this.query(tuple_node)    
  }

  clear() {
    this.consoleTarget.innerHTML = '<span>WhatsitToYou?&nbsp;</span><input data-whatsit-target="input_node" data-action="change->whatsit#search">'
    this.moveCursor()
  }

  moveCursor(){
    const inpts = this.consoleTarget.querySelectorAll('input')
    const last = inpts[inpts.length -1]
    last.focus()
  }
 
  query(tuple){

    const cls = tuple.className
    const val = tuple.value
    const confirm = tuple.dataset.confirm
    const qry = this.queryTarget.value
    this.queryTarget.value = val
    var url
    if (confirm == undefined) {
      url = `/whatsit/new.js?search=${encodeURI(val)}&action_type=${cls}`
    }else {
      url = `/whatsit/new.js?confirm=${encodeURI(confirm)}&action_type=${cls}&resp=${val}`
    }
    Rails.ajax({
      url: url,
      type: "get",
      success: function(data) {
        const viewer = document.getElementById('query_results')
        const last_query = document.getElementById('last_query')
        const inputs = viewer.querySelectorAll('input')
        const inputs_length = inputs.length
        var prev = inputs[inputs_length - 2]
        var last = inputs[inputs_length - 1]
        prev.value = last_query.value
        last.focus()
      }
    })
  }

}